### PR TITLE
jQuery Fathead: Add arguments type & description, Fixes #319

### LINKED ISSUE #319: jQuery Fathead: show examples and args descriptions
## Background

the jQuery IA was recently rewritten by @kotoshenya in #294 and it's already showing good progress, but there's more that can be done to improve it!

For instance we could try showing some code examples.
Also, right now for each method we show which args are required, but it would be more useful if we also explained them, specify what they represent and which values are allowed for each of them, etc.
## To Do

For each article:
- If examples are available, choose a short one (maybe remove the HTML part if it's too long) to show. 
- Show args explanations in the infobox (if possible), instead of the jQuery methods link. Eg "jquery fade in" would show in the infobox, among the other args, "duration: (number or string, default 400) - a string or number determining how long the animation will run"
## Docs

[Learn more about Fatheads](http://docs.duckduckhack.com/resources/fathead-overview.html)
## Forum

See the forum topic [Improve JQuery Fathead](https://forum.duckduckhack.com/t/improve-jquery-fathead/180) for more details.

This issue is part of the [Programming Mission](https://forum.duckduckhack.com/t/duckduckhack-programming-mission-overview/53): help us improve the results for [JavaScript related searches](https://forum.duckduckhack.com/t/javascript-search-overview/94)!

---

IA Page: https://duck.co/ia/view/jquery

### PR DESCRIPTION
**Pull request by thinker3197**
## Description of new Instant Answer, or changes

Added arguments explanation and their types for jQuery fathead

## Related Issues and Discussions

Fixes [#319](https://github.com/duckduckgo/zeroclickinfo-fathead/issues/319)

## People to notify

@bfmags @MariagraziaAlastra 

## Testing & Review
To be completed by Language Leader (or DDG Staff) when reviewing Pull Request

**Code**
- [x] Uniformly indented, well commented
- [x] Fetch.sh and Parse.sh run without errors
- [x] Output contains no blank lines, or multi-line entries

---

Instant Answer Page: https://duck.co/ia/view/jquery

### CODE REVIEW

**Comment by daxtheduck** [comment]
@thinker3197 Thanks for the pull request! Unfortunately, we're only accepting contributions and improvements to a limited portion of the search space.
            
The staff and community are currently focused on making DuckDuckGo the [best search engine for programmers](https://forum.duckduckhack.com/t/duckduckhack-programming-mission-overview/53). To ensure we're making as much progress as possible, **we are only accepting Instant Answers and updates related to the programming mission**.
    
If this is not related to the Programming Mission, it will likely be closed but you can:
    
1. Help us prevent further confusion by telling us where you found out about DuckDuckHack or where we could've alerted you earlier.
2. Keep this code handy! We'll be expanding into other search topics in the future!
3. Join us at our new home: https://forum.duckduckhack.com/

If this is related to the Programming Mission, it will be reviewed by staff and the community as soon as possible!

**Comment by daxtheduck** [comment]
Metadata placeholder post
            
*If you haven't already, please edit your first comment to include a link to the IA page at the bottom.*
            
---
            
**How to find your IA page**
- New Instant Answer
  -- follow the steps [here](https://duck.co/ia/new_ia) to create your IA page.
              
- Existing Instant Answer
  -- find your IA page from this [list](https://duck.co/ia)

**Comment by moollaza** [comment]
@thinker3197 thanks for the awesome work! This is looking great.

I left a couple comments about a few redirects that looked odd.

I noticed you're using a code block to list the params. We should be using a `<ul>` instead. The content isn't code, so we don't need to try and syntax highlight it.

After you switch to use an unordered list, and remove the duplicates this should be good for merge 👍 

Thanks again 😄

File: lib/fathead/jquery/output.txt [hunk 1]
```diff
@@ -1,1919 +1,1919 @@
-.add()	A			jQuery Traversing\njQuery Miscellaneous Traversal							<section class="prog__container"><p>Create a new jQuery object with elements added to the set of matched elements.</p><pre><code>$(selector).add(selector) → jQuery\n$(selector).add(elements) → jQuery\n$(selector).add(html) → jQuery\n$(selector).add(selection) → jQuery\n$(selector).add(selector, context) → jQuery</code></pre></section>	https://api.jquery.com/add/
-.addBack()	A			jQuery Traversing\njQuery Miscellaneous Traversal							<section class="prog__container"><p>Add the previous set of elements on the stack to the current set, optionally filtered by a selector.</p><pre><code>$(selector).addBack(selector) → jQuery</code></pre></section>	https://api.jquery.com/addBack/
+	R	jQuery()										
+$.)	R	jQuery()
```
Comment (thinker3197) [inline lib/fathead/jquery/output.txt]: Seems like an issue with the entries, it's there in the current version also. 
https://github.com/duckduckgo/zeroclickinfo-fathead/blob/master/lib/fathead/jquery/output.txt#L1652

**Comment by moollaza** [comment]
@thinker3197 Thanks! This LGTM :+1:

Excellent work! 💯

File: lib/fathead/jquery/parse.js [hunk 2]
```diff
@@ -40,13 +41,25 @@ if (require.main === module) {
...
+          const argsDesc = entry.argsDesc
+                            .filter((item, pos, self) => self.indexOf(item) == pos)
 
-          return `<p>${entry.description}</p>` +
+          let desc = `<p>${entry.description}</p>` +
             `<pre><code>${signatures.join('\n')}</code></pre>`
-        })
 
+          if(argsDesc.length && argsDesc[0] !== '') {
+            desc += `<h4>Arguments</h4>` +
```
Comment (moollaza) [inline lib/fathead/jquery/parse.js]: Whoops, please use `<span class="prog__sub">` -- this is outlined in the docs

File: lib/fathead/jquery/output.txt [hunk 3]
```diff
@@ -1,1919 +1,1919 @@
-.add()	A			jQuery Traversing\njQuery Miscellaneous Traversal							<section class="prog__container"><p>Create a new jQuery object with elements added to the set of matched elements.</p><pre><code>$(selector).add(selector) → jQuery\n$(selector).add(elements) → jQuery\n$(selector).add(html) → jQuery\n$(selector).add(selection) → jQuery\n$(selector).add(selector, context) → jQuery</code></pre></section>	https://api.jquery.com/add/
-.addBack()	A			jQuery Traversing\njQuery Miscellaneous Traversal							<section class="prog__container"><p>Add the previous set of elements on the stack to the current set, optionally filtered by a selector.</p><pre><code>$(selector).addBack(selector) → jQuery</code></pre></section>	https://api.jquery.com/addBack/
+	R	jQuery()
```
Comment (moollaza) [inline lib/fathead/jquery/output.txt]: @thinker3197 can you ensure we check that the title has a length before creating an entry for it?

Comment (thinker3197) [inline lib/fathead/jquery/output.txt]: Done. It was an interesting issue though. The parser adds redirects for each jQuery method by removing parenthesis, `jQuery` symbol and prepending `$` to the method name. However, when it comes the core jQuery function (which is named `jQuery`), dollarizing it and removing `jQuery` symbol creates an empty title. 
Fixed it by applying a simple filter to check for alphanumerics in title  🎉

**Comment by moollaza** [comment]
@thinker3197 Thanks! This LGTM :+1:

**Pull request merged**